File: Java/ChatterBox/src/data/ChatUser.java
package data;

import java.awt.Color;
import java.net.InetAddress;

public class ChatUser implements IUser
{
    private String name;
    private String displayName;
    private Color displayColor;
    private boolean available;

    /***************************************************************************
     * Constructor
     * 
     * @param displayName
     * @param isAvailable
     **************************************************************************/
    public ChatUser( String displayName, boolean isAvailable )
    {
        this( displayName, "255", isAvailable );
    }

    /***************************************************************************
     * Constructor
     * 
     * @param displayName
     * @param colorRGB
     * @param isAvailable
     **************************************************************************/
    public ChatUser( String displayName, String colorRGB, boolean isAvailable )
    {
        try
        {
            this.name = System.getProperty( "user.name" )
                    + InetAddress.getLocalHost().getHostName() + "@";
        } catch( Exception e )
        {
            this.name = System.getProperty( "user.name" ) + "@unknown.host";
        }

        this.displayName = displayName;
        this.available = isAvailable;
        try
        {
            displayColor = new Color( Integer.parseInt( colorRGB ) );
        } catch( Exception e )
        {
            displayColor = Color.black;
        }
    }

    /*
     * (non-Javadoc)
     * 
     * @see data.IUser#getName()
     */
    @Override
    public String getName()
    {
        return name;
    }

    /*
     * (non-Javadoc)
     * 
     * @see data.IUser#getDisplayName()
     */
    @Override
    public String getDisplayName()
    {
        return displayName;
    }

    /*
     * (non-Javadoc)
     * 
     * @see data.IUser#setDisplayName(java.lang.String)
     */
    @Override
    public void setDisplayName( String displayName )
    {
        this.displayName = displayName;
    }

    /*
     * (non-Javadoc)
     * 
     * @see data.IUser#getDisplayColor()
     */
    @Override
    public Color getDisplayColor()
    {
        return displayColor;
    }

    /*
     * (non-Javadoc)
     * 
     * @see data.IUser#setDisplayColor(java.awt.Color)
     */
    @Override
    public void setDisplayColor( Color color )
    {
        displayColor = color;
    }

    /*
     * (non-Javadoc)
     * 
     * @see data.IUser#isAvailable()
     */
    @Override
    public boolean isAvailable()
    {
        return available;
    }

    /*
     * (non-Javadoc)
     * 
     * @see data.IUser#setAvailable(boolean)
     */
    @Override
    public void setAvailable( boolean available )
    {
        this.available = available;
    }
}

File: C#/ChatterBox/ChatterBox/ChatUser.cs
﻿using System.Security.Principal;
using System.Drawing;

namespace ChatterBox
{
    /// <summary>
    /// This class represents a chat user.
    /// </summary>
    class ChatUser : IUser
    {
        private string name;
        private string displayName;
        private Color preferredColor;
        private bool available;

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="displayName"></param>
        public ChatUser( string displayName ) : this( displayName, Color.Black )
        {
        }

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="displayName"></param>
        /// <param name="preferredColor"></param>
        public ChatUser( string displayName, Color preferredColor )
        {
            this.name = WindowsIdentity.GetCurrent().Name; ;
            this.displayName = displayName;
            this.preferredColor = preferredColor;
            available = true;
        }

        /// <summary cref="IUser.getName">
        /// <see cref="IUser.getName"/>
        /// </summary>
        /// <returns></returns>
        public string getName()
        {
            return name;
        }

        /// <summary cref="IUser.getDisplayName">
        /// <see cref="IUser.getDisplayName"/>
        /// </summary>
        /// <returns></returns>
        public string getDisplayName()
        {
            return displayName;
        }

        /// <summary cref="IUser.setDisplayName">
        /// <see cref="IUser.setDisplayName"/>
        /// </summary>
        /// <param name="displayName"></param>
        public void setDisplayName( string displayName )
        {
            this.displayName = displayName;
        }

        /// <summary cref="IUser.isAvailable">
        /// <see cref="IUser.isAvailable"/>
        /// </summary>
        /// <returns></returns>
        public bool isAvailable()
        {
            return available;
        }

        /// <summary cref="IUser.setAvailable">
        /// <see cref="IUser.setAvailable"/>
        /// </summary>
        /// <param name="available"></param>
        public void setAvailable( bool available )
        {
            this.available = available;
        }

        /// <summary>
        /// Returns this user's preferred color.
        /// </summary>
        /// <returns></returns>
        public Color getPreferredColor()
        {
            return preferredColor;
        }

        /// <summary>
        /// Sets this user's preferred color.
        /// </summary>
        /// <param name="color"></param>
        public void setPreferredColor( Color color )
        {
            preferredColor = color;
        }
    }
}
